pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -63,34 +93,81 @@ let
 
   meta = with lib; {
     homepage = "https://github.com/bazelbuild/bazel/";
     description = "Build tool that builds code quickly and reliably";
     license = licenses.asl20;
     maintainers = [ maintainers.mboes ];
-    platforms = platforms.linux ++ platforms.darwin;
+    inherit platforms;
   };
 
   # Additional tests that check bazel’s functionality. Execute
   #
   #     nix-build . -A bazel.tests
   #
   # in the nixpkgs checkout root to exercise them locally.
-  passthru.tests = {
-    pythonBinPath = callPackage ./python-bin-path-test.nix {};
-    bashTools = callPackage ./bash-tools-test.nix {};
-  };
+  passthru.tests =
+    let
+      runLocal = name: attrs: script: runCommandCC name ({
+        preferLocalBuild = true;
+        meta.platforms = platforms;
+      } // attrs) script;
+
+      # bazel wants to extract itself into $install_dir/install every time it runs,
+      # so let’s do that only once.
+      extracted =
+        let install_dir =
+          # `install_base` field printed by `bazel info`, minus the hash.
+          # yes, this path is kinda magic. Sorry.
+          if stdenv.hostPlatform.isDarwin
+          then "/var/tmp/_bazel_nixbld"
+          else "$HOME/.cache/bazel/_bazel_nixbld";
+        in runLocal "bazel-extracted-homedir" { passthru.install_dir = install_dir; } ''
+            export HOME=$(mktemp -d)
+            ${bazel}/bin/bazel version
+            cp -R ${install_dir} $out
+          '';
+
+      bazelTest = { name, bazelScript, workspaceDir }:
+        runLocal name {} ''
+          # set up home with pre-unpacked bazel
+          export HOME=$(mktemp -d)
+          mkdir -p ${extracted.install_dir}
+          cp -R ${extracted}/install ${extracted.install_dir}
+
+          # https://stackoverflow.com/questions/47775668/bazel-how-to-skip-corrupt-installation-on-centos6
+          # Bazel checks whether the mtime of the install dir files
+          # is >9 years in the future, otherwise it extracts itself again.
+          # see PosixFileMTime::IsUntampered in src/main/cpp/util
+          # What the hell bazel.
+          ${lr}/bin/lr -0 -U ${extracted.install_dir} | ${xe}/bin/xe -N0 -0 touch --date="9 years 6 months" {}
+
+          # Note https://github.com/bazelbuild/bazel/issues/5763#issuecomment-456374609
+          # about why to create a subdir for the workspace.
+          cp -r ${workspaceDir} wd && chmod u+w wd && cd wd
+
+          ${bazelScript}
+
+          touch $out
+        '';
+
+    in {
+      pythonBinPath = callPackage ./python-bin-path-test.nix{ inherit runLocal bazelTest; };
+      bashTools = callPackage ./bash-tools-test.nix { inherit runLocal bazelTest; };
+    };
 
   name = "bazel-${version}";
 
   src = fetchurl {
     url = "https://github.com/bazelbuild/bazel/releases/download/${version}/${name}-dist.zip";
-    sha256 = "11gsc00ghxqkbci8nrflkwq1lcvqawlgkaryj458b24si6bjl7b2";
+    sha256 = "d26dadf62959255d58e523da3448a6222af768fe1224e321b120c1d5bbe4b4f2";
```

Comment:
Are we standardizing on a different format for the hash?

Reply:
Hashes can have multiple encodings, This looks like the first one is baze32 while the second is base16

Reply:
Nix's standard is base32, at least for now. See https://github.com/NixOS/nix/issues/806#issuecomment-475191490